src/main/java/org/reactome/addlinks/dataretrieval/pharos/PharosLigandDataRetriever.java

```diff
@@ -0,0 +1,78 @@
+package org.reactome.addlinks.dataretrieval.pharos;
+
+import java.io.FileWriter;
+import java.io.IOException;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class PharosLigandDataRetriever extends PharosDataRetriever
+{
+	// NOTE: It might be tempting to add "\n" at the end of the "lines" in this string below, but that will result in a "Bad request" response
+	// from the GraphQL service. Clearly, queries cannot contain real line breaks.
+	// we will create cross-references based on GtP identifiers.
+	private static final String LIGANDS_QUERY = "{ \"query\": \"query {"
+			+ "  ligands(filter: {facets: [{facet:\\\"Data Source\\\" values:[\\\"Guide to Pharmacology\\\"]}]}){"
+			+ "    count"
+			+ "    ligands(top:"+PharosDataRetriever.TOP_AMOUNT_TOKEN+" skip:"+PharosDataRetriever.SKIP_AMOUNT_TOKEN+"){"
+			+ "      ligid"
+			+ "      name"
+			+ "      isdrug"
+			+ "      synonyms {"
+			+ "        name"
+			+ "        value"
+			+ "      }"
+			+ "    }"
+			+ "  }"
+			+ "}\"}";
+
+	@Override
+	protected int processJSONArray(FileWriter fw, JSONObject jsonObj) throws IOException
+	{
+		try
+		{
+			JSONArray ligands = (((JSONObject)((JSONObject) jsonObj.get("data")).get("ligands")).getJSONArray("ligands"));
+			for (int i = 0; i < ligands.length(); i++)
+			{
+				JSONObject ligand = ((JSONObject)ligands.get(i));
+				Boolean isDrug = ligand.getBoolean("isdrug");
+				// Check that the ligand has "isdrug==true" and then get the GtP synonym's value (synonym's "name" will be "Guide to Pharmacology").
+				if (isDrug.booleanValue())
```

Comment:
Could line 39 be `boolean isDrug = ligand.getBoolean("isdrug")`?  Java will automatically unbox the object Boolean to a primitive and that way line 41 could just be `if(isDrug)`.

Reply:
Hmm I can try that.